beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -211,13 +211,6 @@ func (s *Service) validateBeaconBlock(ctx context.Context, blk interfaces.Signed
 		return err
 	}
 
-	hasStateSummaryDB := s.cfg.beaconDB.HasStateSummary(ctx, bytesutil.ToBytes32(blk.Block().ParentRoot()))
-	if !hasStateSummaryDB {
```

Comment:
Are we sure all paths (init sync and regular sync) guarantee that this code is not hit?